### PR TITLE
Day 2: Add Bare Metal Host feature

### PR DESCRIPTION
**Pull request by mareklibra**
The user is able to add 1+ worker nodes in OCM.

OCM counter-part:  https://gitlab.cee.redhat.com/service/uhc-portal/-/merge_requests/1748

--- 

Note for UX: video will follow

### CODE REVIEW

File: src/components/AddBareMetalHosts/AddBareMetalHostsDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        }
+
+        if (!dayTwoClusterExists) {
+          try {
+            // Optionally create Day 2 cluster
+            const { data } = await addHostsClusters({
+              id: clusterId,
+              name: cluster.name,
+              openshiftVersion,
+              apiVipDnsname,
```
Comment (jtomasek) [inline src/components/AddBareMetalHosts/AddBareMetalHostsDialog.tsx:84]: Correcting these small spelling errors on API side will be quite hard to do later :/

**Review by jtomasek — CHANGES_REQUESTED** [review]
There is a problem when creating `addHostCluster` from OCM cluster which uses UUID as a name as the AI API does not allow cluster name to begin with a number. Changing the 'display name' of the cluster did not help.

File: src/api/addHostsCluster.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { AxiosPromise } from 'axios';
+import { client } from './axiosClient';
+import { AddHostsClusterCreateParams, Cluster } from './types';
+
+export const addHostsClusters = (params: AddHostsClusterCreateParams): AxiosPromise<Cluster> =>
+  client.post('/add_hosts_clusters', params);
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply src/api/addHostsCluster.ts:6]: IIUC, the issue is just in naming of the endpoint. Let's move this discussion to slack and react in UI (follow-up) accordingly.
But you are right, there will never be better time to change this than now.

File: src/components/AddBareMetalHosts/AddBareMetalHostsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  return (
+    <Modal
+      aria-label="Add Bare Metal Hosts dialog"
+      title="Add Bare Metal Hosts"
+      isOpen={true}
+      onClose={closeModal}
+      variant={ModalVariant.small}
+    >
+      {error && (
+        <Alert isInline variant={AlertVariant.danger} title="Add Bare Metal Hosts failed">
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply src/components/AddBareMetalHosts/AddBareMetalHostsDialog.tsx:117]: done

File: src/components/hosts/RoleCell.tsx [hunk 4]
```diff
@@ -6,10 +6,11 @@ import { canEditRole, getHostRole } from './utils';
 type RoleCellProps = {
   host: Host;
   clusterStatus: Cluster['status'];
+  forceStatic?: boolean;
 };
 
-const RoleCell: React.FC<RoleCellProps> = ({ host, clusterStatus }) => {
-  return canEditRole(clusterStatus, host.status) ? (
+const RoleCell: React.FC<RoleCellProps> = ({ host, clusterStatus, forceStatic = false }) => {
+  return !forceStatic && canEditRole(clusterStatus, host.status) ? (
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply src/components/hosts/RoleCell.tsx:13]: right, it is not needed. And recent canEditRole is already false for this clusterStatus. So: done.

File: src/components/hosts/HostsTable.tsx [hunk 5]
```diff
@@ -95,7 +101,13 @@ const hostToHostTableRow = (openRows: OpenRows, cluster: Cluster) => (host: Host
           sortableValue: computedHostname || '',
         },
         {
-          title: <RoleCell host={host} clusterStatus={cluster.status} />,
+          title: (
+            <RoleCell
+              host={host}
+              clusterStatus={cluster.status}
+              forceStatic={variant != 'Cluster'}
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply src/components/hosts/HostsTable.tsx:108]: done

File: src/components/AddBareMetalHosts/BareMetalInventoryAddHost.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  const [isDiscoveryHintModalOpen, setDiscoveryHintModalOpen] = React.useState(false);
+
+  return (
+    <>
+      <TextContent>
+        <Text component="h2">Bare Metal Inventory</Text>
+        <Text component="p">
+          <DiscoveryImageModalButton ButtonComponent={Button} cluster={cluster} />
+        </Text>
+        <Text component="p">
```
Comment (mareklibra) [inline src/components/AddBareMetalHosts/BareMetalInventoryAddHost.tsx:30]: These are adepts for reusable components but contemplating if it is not over-killing.

**Review by jtomasek — APPROVED** [review]
There are several changes we should do as a follow up:
1. Render the day2 cluster as a tab within the OCM cluster detail page:
    - Remove AddBareMetalHost modal and instead the action would activate the 'scale cluster' or 'add hosts' tab in OCM cluster detail. Here we'd fetch the data and create/load the AI day 2 cluster resource.
    - Allows us to avoid showing day2 clusters from the AI clusters list
2. Implement the same (as 1.) day2 integration for AI day1 cluster
3. Backend should update the status_info for cluster in 'adding_hosts' state to something like: 'This cluster is used for adding hosts to existing cluster <name/id>
4. Improve description of 'added-to-existing-cluster' cluster status to 'Host has booted RHCOS image, please use cluster console to monitor further progress of adding this host to the cluster'

**Comment by mareklibra** [comment]
@jtomasek , thanks for your comments. Avoiding modal dialog in favour of new OCM detail subtab will make the integration tighter and more user friendly. I believe transition from this PR state to the subtab will not be complicated.

**Pull request merged**